include/materials/norsand.tcc

```diff
@@ -35,19 +35,35 @@ mpm::NorSand<Tdim>::NorSand(unsigned id, const Json& material_properties)
     p_image_initial_ =
         material_properties.at("p_image_initial").template get<double>();
     // Flag for bonded model
-    bond_model_ = material_properties.at("bond_model").template get<bool>();
-    // Initial p_cohesion
-    p_cohesion_initial_ =
-        material_properties.at("p_cohesion_initial").template get<double>();
-    // Initial p_dilation
-    p_dilation_initial_ =
-        material_properties.at("p_dilation_initial").template get<double>();
-    // Cohesion degradation parameter m upon shearing
-    m_cohesion_ = material_properties.at("m_cohesion").template get<double>();
-    // Dilation degradation parameter m upon shearing
-    m_dilation_ = material_properties.at("m_dilation").template get<double>();
-    // Parameter for shear modulus
-    m_modulus_ = material_properties.at("m_modulus").template get<double>();
+    if (material_properties.find("bond_model") != material_properties.end())
+      bond_model_ = material_properties.at("bond_model").template get<bool>();
+
+    // Obtain the rest of the bonded parameters
+    if (bond_model_) {
+      // Initial p_cohesion
+      p_cohesion_initial_ =
+          material_properties.at("p_cohesion_initial").template get<double>();
+      // Initial p_dilation
+      p_dilation_initial_ =
+          material_properties.at("p_dilation_initial").template get<double>();
+      // Cohesion degradation parameter m upon shearing
+      m_cohesion_ = material_properties.at("m_cohesion").template get<double>();
+      // Dilation degradation parameter m upon shearing
+      m_dilation_ = material_properties.at("m_dilation").template get<double>();
+      // Parameter for shear modulus
+      m_modulus_ = material_properties.at("m_modulus").template get<double>();
+    } else {
+      // Initial p_cohesion
```

Comment:
Could we set these as the default values in the header and we don't need an else clause